prov/efa/test/efa_unit_test_common.c

```diff
@@ -22,20 +23,81 @@ int efa_unit_test_resource_construct(struct efa_resource* resource)
 		return ret;
 	}
 
+	ret = fi_endpoint(resource->domain, resource->info, &resource->ep, NULL);
+	if (ret) {
+		fi_close(&resource->domain->fid);
```

Comment:
It seems we can call `efa_unit_test_resource_destroy` on the error handling path?

Reply:
Good point. Will address.